SafeguardPasswords/Src/Modules/Request/Request.pqm

```diff
@@ -0,0 +1,147 @@
+let  
+    ValidateByStatusCode = Extension.ImportFunction("ValidateByStatusCode", "ResponseHandler.pqm"),
+    InfoLog = Extension.ImportFunction("InfoLog", "CommonLogger.pqm"),
+    ErrorCodes = Extension.ImportFunction("ErrorCodes", "RequestErrors.pqm"),
+    GetHost = Extension.ImportFunction("GetSppHost","UrlBuilder.pqm"),
+    Constants = Extension.ImportModule("Constants.pqm")[SppConstants],
+    IsMocked = Extension.ImportFunction("IsMocked", "CommonUtils.pqm"),
+
+
+    Request.GetSppResponseBase = (host as text, options as record, optional logResponse as logical, optional mocks as record) =>
+        let
+
+            // IsRetry causes Web.Contents to not use cached API response. 
+            // See https://learn.microsoft.com/en-us/powerquery-m/web-contents for details
+            response = if IsMocked(mocks, "Response")
+                        then mocks[Response]
+                        else Web.Contents(host, options & [ManualStatusHandling = {400,406}, IsRetry = true]),
```

Comment:
Do we only want to handle the 400 and 406 errors manually? The others are not necessary, such as 404? Shouldn't the ErrorCodes from CommonRequestErrors.pqm be used here?

Reply:
SPP provides clear API error responses in almost all cases and I found them to be clear enough during development. There were two exceptions - 406 and 400 (specifically when returned from the RSTS). So for the SPP connector, those are the only two I manually handled.

As mentioned earlier, RSTS returns a 400 when invalid credentials are presented. I needed to capture this error and return something sensible.

Reply:
ACK.